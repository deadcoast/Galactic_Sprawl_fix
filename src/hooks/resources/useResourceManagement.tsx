import { useCallback, useEffect, useMemo, useState } from 'react';
import { resourceManager } from '../../managers/game/ResourceManager';
import {
  ResourceIntegration,
  createResourceIntegration,
} from '../../managers/resource/ResourceIntegration';
import { ResourceState, ResourceType } from '../../types/resources/ResourceTypes';

// Singleton instance of the resource integration
let resourceIntegrationInstance: ResourceIntegration | null = null;

/**
 * Initialize the resource integration if it hasn't been initialized yet
 */
function getResourceIntegration(): ResourceIntegration {
  if (!resourceIntegrationInstance) {
    resourceIntegrationInstance = createResourceIntegration(resourceManager);
  }

  if (!resourceIntegrationInstance) {
    throw new Error('Failed to create resource integration instance');
  }

  return resourceIntegrationInstance;
}

/**
 * Hook for accessing the resource management system
 */
export function useResourceManagement() {
  const [resourceStates, setResourceStates] = useState<Map<ResourceType, ResourceState>>(new Map());
  const [isInitialized, setIsInitialized] = useState(false);

  // Get or create the resource integration
  const integration = useMemo(() => getResourceIntegration(), []);

  // Initialize the resource states
  useEffect(() => {
    // Get all resource types
    const resourceTypes = Array.from(resourceManager['resources'].keys()) as ResourceType[];

    // Create a map of resource states
    const states = new Map<ResourceType, ResourceState>();
    resourceTypes.forEach(type => {
      const state = resourceManager.getResourceState(type);
      if (state) {
        states.set(type, state);
      }
    });

    setResourceStates(states);
    setIsInitialized(true);
  }, []);

  // Update the resource integration on each frame
  useEffect(() => {
    if (!isInitialized) {
      return;
    }

    const frameId = requestAnimationFrame(time => {
      integration.update(1000 / 60); // Assume 60 FPS
    });

    return () => {
      cancelAnimationFrame(frameId);
    };
  }, [integration, isInitialized]);

  // Clean up the resource integration when the component unmounts
  useEffect(() => {
    return () => {
      if (resourceIntegrationInstance) {
        resourceIntegrationInstance.cleanup();
        resourceIntegrationInstance = null;
      }
    };
  }, []);

  // Get a resource state
  const getResourceState = useCallback(
    (type: ResourceType): ResourceState | undefined => {
      return resourceStates.get(type);
    },
    [resourceStates]
  );

  // Get all resource states
  const getAllResourceStates = useCallback((): Map<ResourceType, ResourceState> => {
    return resourceStates;
  }, [resourceStates]);

  // Get resource amount
  const getResourceAmount = useCallback(
    (type: ResourceType): number => {
      return resourceStates.get(type)?.current || 0;
    },
    [resourceStates]
  );

  // Check if a resource is available
  const hasResource = useCallback(
    (type: ResourceType, amount: number): boolean => {
      const state = resourceStates.get(type);
      return state ? state.current >= amount : false;
    },
    [resourceStates]
  );

  // Check if multiple resources are available
  const hasResources = useCallback(
    (resources: Array<{ type: ResourceType; amount: number }>): boolean => {
      return resources.every(({ type, amount }) => hasResource(type, amount));
    },
    [hasResource]
  );

  // Consume a resource
  const consumeResource = useCallback(
    (type: ResourceType, amount: number): boolean => {
      if (!hasResource(type, amount)) {
        return false;
      }

      resourceManager.removeResource(type, amount);
      return true;
    },
    [hasResource]
  );

  // Consume multiple resources
  const consumeResources = useCallback(
    (resources: Array<{ type: ResourceType; amount: number }>): boolean => {
      if (!hasResources(resources)) {
        return false;
      }

      resources.forEach(({ type, amount }) => {
        resourceManager.removeResource(type, amount);
      });

      return true;
    },
    [hasResources]
  );

  // Add a resource
  const addResource = useCallback((type: ResourceType, amount: number): void => {
    resourceManager.addResource(type, amount);
  }, []);

  // Add multiple resources
  const addResources = useCallback(
    (resources: Array<{ type: ResourceType; amount: number }>): void => {
      resources.forEach(({ type, amount }) => {
        resourceManager.addResource(type, amount);
      });
    },
    []
  );

  // Get resource production rate
  const getProductionRate = useCallback(
    (type: ResourceType): number => {
      return resourceStates.get(type)?.production || 0;
    },
    [resourceStates]
  );

  // Get resource consumption rate
  const getConsumptionRate = useCallback(
    (type: ResourceType): number => {
      return resourceStates.get(type)?.consumption || 0;
    },
    [resourceStates]
  );

  // Set resource production rate
  const setProductionRate = useCallback((type: ResourceType, rate: number): void => {
    resourceManager.setResourceProduction(type, rate);
  }, []);

  // Set resource consumption rate
  const setConsumptionRate = useCallback((type: ResourceType, rate: number): void => {
    resourceManager.setResourceConsumption(type, rate);
  }, []);

  // Get resource capacity
  const getResourceCapacity = useCallback(
    (type: ResourceType): number => {
      return resourceStates.get(type)?.max || 0;
    },
    [resourceStates]
  );

  // Get resource percentage
  const getResourcePercentage = useCallback(
    (type: ResourceType): number => {
      const state = resourceStates.get(type);
      if (!state || state.max === 0) {
        return 0;
      }
      return (state.current / state.max) * 100;
    },
    [resourceStates]
  );

  return {
    // Resource state
    getResourceState,
    getAllResourceStates,
    getResourceAmount,
    hasResource,
    hasResources,

    // Resource modification
    consumeResource,
    consumeResources,
    addResource,
    addResources,

    // Resource rates
    getProductionRate,
    getConsumptionRate,
    setProductionRate,
    setConsumptionRate,

    // Resource capacity
    getResourceCapacity,
    getResourcePercentage,

    // Initialization state
    isInitialized,
  };
}
