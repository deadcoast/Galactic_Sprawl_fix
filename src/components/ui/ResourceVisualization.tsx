import { AlertTriangle, Beaker, Database, Users, Zap } from 'lucide-react';
import * as React from "react";
import { ResourceType } from "./../../types/resources/ResourceTypes";
import { resourceTypeToString } from '../../utils/ResourceTypeConverter';
import { useTooltipContext } from './tooltip-context';

interface ResourceVisualizationProps {
  resourceType: ResourceType | string;
  amount: number;
  showIcon?: boolean;
  showAmount?: boolean;
  showName?: boolean;
  size?: 'small' | 'medium' | 'large';
  className?: string;
  onClick?: () => void;
}

/**
 * ResourceVisualization Component
 *
 * Displays a resource with its icon, amount, and name.
 * This component is designed to work with both string and enum resource types
 * to facilitate the transition from string-based to enum-based resource types.
 */
const ResourceVisualization: React.FC<ResourceVisualizationProps> = ({
  resourceType,
  amount,
  showIcon = true,
  showAmount = true,
  showName = false,
  size = 'medium',
  className = '',
  onClick,
}) => {
  // Convert the resource type to string format for consistent handling
  const stringType =
    typeof resourceType === 'string' ? resourceType : resourceTypeToString(resourceType);

  // Convert to lowercase for map lookups (for backward compatibility)
  const lowerType = stringType.toLowerCase();

  // Determine the size of the component
  const sizeClasses = {
    small: 'text-xs',
    medium: 'text-sm',
    large: 'text-base',
  };

  // Resource icon mapping
  const iconComponents = {
    minerals: Database,
    energy: Zap,
    population: Users,
    research: Beaker,
  };

  // Resource color mapping
  const colors = {
    minerals: '#8884d8',
    energy: '#ffc658',
    population: '#82ca9d',
    research: '#8dd1e1',
  };

  // Resource name mapping
  const names = {
    minerals: 'Minerals',
    energy: 'Energy',
    population: 'Population',
    research: 'Research',
  };

  // Get the appropriate icon based on the resource type
  const getIcon = () => {
    const IconComponent = iconComponents[lowerType as keyof typeof iconComponents] || AlertTriangle;
    return <IconComponent className="h-4 w-4" />;
  };

  // Get the appropriate color based on the resource type
  const getColor = () => {
    return colors[lowerType as keyof typeof colors] || '#999999';
  };

  // Get the appropriate name based on the resource type
  const getName = () => {
    return names[lowerType as keyof typeof names] || stringType;
  };

  // Get the tooltip context
  const { showTooltip, hideTooltip } = useTooltipContext();

  // Create a ref for the component
  const componentRef = React.useRef<HTMLDivElement>(null);

  // Handle mouse enter event
  const handleMouseEnter = () => {
    if (componentRef.current) {
      const rect = componentRef.current.getBoundingClientRect();
      showTooltip(
        <div className="rounded bg-gray-800 p-2 text-white shadow-lg">
          <div className="font-bold">{getName()}</div>
          <div>Amount: {amount}</div>
        </div>,
        { x: rect.left + rect.width / 2, y: rect.top }
      );
    }
  };

  // Handle mouse leave event
  const handleMouseLeave = () => {
    hideTooltip();
  };

  return (
    <div
      ref={componentRef}
      className={`resource-visualization flex items-center ${sizeClasses[size]} ${className}`}
      onMouseEnter={handleMouseEnter}
      onMouseLeave={handleMouseLeave}
      onClick={onClick}
      style={{ cursor: onClick ? 'pointer' : 'default' }}
    >
      {showIcon && (
        <div
          className="resource-icon mr-1 flex h-6 w-6 items-center justify-center rounded-full"
          style={{ backgroundColor: getColor() }}
        >
          {getIcon()}
        </div>
      )}
      {showAmount && <div className="resource-amount mr-1">{amount}</div>}
      {showName && <div className="resource-name">{getName()}</div>}
    </div>
  );
};

export default ResourceVisualization;
