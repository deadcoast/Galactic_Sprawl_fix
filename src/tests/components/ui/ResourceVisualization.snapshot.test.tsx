import React from "react";
import { render, screen } from '@testing-library/react';
import { createContext, Dispatch, ReactNode, useCallback, useReducer } from 'react';
import { describe, expect, it, vi } from 'vitest';
import ResourceVisualization from '../../../components/ui/ResourceVisualization';
import { ResourceType } from "./../../../types/resources/ResourceTypes";
import { ResourceType } from "./../../../types/resources/ResourceTypes";

// Create a mock GameContext since we can't import it directly
interface GameContextType {
  state: TestGameState;
  dispatch: Dispatch<TestGameAction>;
  updateShip: () => void;
  addMission: () => void;
  updateSector: () => void;
}

const GameContext = createContext<GameContextType | null>(null);

// Define our own minimal types to avoid conflicts with actual GameContext types
interface ResourceState {
  [ResourceType.MINERALS]: number;
  [ResourceType.ENERGY]: number;
  [ResourceType.POPULATION]: number;
  [ResourceType.RESEARCH]: number;
}

interface TestGameState {
  resources: ResourceState;
  resourceRates: ResourceState;
}

type TestGameAction = {
  type: 'UPDATE_RESOURCES';
  resources: Partial<ResourceState>;
};

// Define a type that matches the GameContext value structure but with our test types
interface TestGameContextValue {
  state: TestGameState;
  dispatch: Dispatch<TestGameAction>;
  updateShip: () => void;
  addMission: () => void;
  updateSector: () => void;
}

// Create a simplified test provider that focuses only on what ResourceVisualization needs
function TestGameProvider({
  children,
  initialResources = {
    [ResourceType.MINERALS]: 1000,
    [ResourceType.ENERGY]: 1000,
    [ResourceType.POPULATION]: 100,
    [ResourceType.RESEARCH]: 0,
  },
}: {
  children: ReactNode;
  initialResources?: ResourceState;
}) {
  // Create a simplified reducer that only handles resource updates
  const reducer = (state: TestGameState, action: TestGameAction): TestGameState => {
    switch (action.type) {
      case 'UPDATE_RESOURCES':
        return {
          ...state,
          resources: { ...state.resources, ...action.resources },
        };
      default:
        return state;
    }
  };

  // Create a minimal state that only includes what the ResourceVisualization needs
  const [state, dispatch] = useReducer(reducer, {
    resources: initialResources,
    resourceRates: {
      [ResourceType.MINERALS]: 0,
      [ResourceType.ENERGY]: 0,
      [ResourceType.POPULATION]: 0,
      [ResourceType.RESEARCH]: 0,
    },
  });

  // Create stub implementations for the other context methods that aren't used by ResourceVisualization
  const updateShip = useCallback(() => {}, []);
  const addMission = useCallback(() => {}, []);
  const updateSector = useCallback(() => {}, []);

  // Create a context value that looks like what GameContext would provide
  // but is actually just a simple object with the minimal properties needed
  const contextValue: TestGameContextValue = {
    state,
    dispatch,
    updateShip,
    addMission,
    updateSector,
  };

  // Use type assertion to bypass type checking - necessary for testing
  return <GameContext.Provider value={contextValue}>{children}</GameContext.Provider>;
}

// Mock the useGameState hook that ResourceVisualization uses
vi.mock('../../../contexts/GameContext', () => ({
  useGameState: (selector: (state: TestGameState) => unknown) => {
    // This mock implementation assumes the selector is extracting resources or resourceRates
    return {
      [ResourceType.MINERALS]: 1000,
      [ResourceType.ENERGY]: 1000,
      [ResourceType.POPULATION]: 100,
      [ResourceType.RESEARCH]: 0,
    };
  },
}));

describe('ResourceVisualization Component', () => {
  it('renders with default resource values', () => {
    render(
      <TestGameProvider>
        <ResourceVisualization />
      </TestGameProvider>
    );

    // Check for resource labels
    expect(screen.getByText(ensureStringResourceType(ResourceType.MINERALS))).toBeInTheDocument();
    expect(screen.getByText(ensureStringResourceType(ResourceType.ENERGY))).toBeInTheDocument();
    expect(screen.getByText(ensureStringResourceType(ResourceType.POPULATION))).toBeInTheDocument();
    expect(screen.getByText(ensureStringResourceType(ResourceType.RESEARCH))).toBeInTheDocument();

    // Use getAllByText for values that appear multiple times
    const mineralValues = screen.getAllByText(/1,000/, { exact: false });
    expect(mineralValues.length).toBeGreaterThan(0);

    const populationValue = screen.getByText('100');
    expect(populationValue).toBeInTheDocument();

    const researchValue = screen.getByText('0');
    expect(researchValue).toBeInTheDocument();
  });

  it('renders with low resource warning', () => {
    render(
      <TestGameProvider
        initialResources={{
          [ResourceType.MINERALS]: 900, // Just below the low threshold (1000)
          [ResourceType.ENERGY]: 1000,
          [ResourceType.POPULATION]: 100,
          [ResourceType.RESEARCH]: 0,
        }}
      >
        <ResourceVisualization />
      </TestGameProvider>
    );

    // Check for low minerals warning
    expect(
      screen.getByText(`Low ${ensureStringResourceType(ResourceType.MINERALS)} levels`)
    ).toBeInTheDocument();
  });

  it('renders with critical resource warning', () => {
    render(
      <TestGameProvider
        initialResources={{
          [ResourceType.MINERALS]: 400, // Below the critical threshold (500)
          [ResourceType.ENERGY]: 300, // Below the critical threshold (400)
          [ResourceType.POPULATION]: 100,
          [ResourceType.RESEARCH]: 0,
        }}
      >
        <ResourceVisualization />
      </TestGameProvider>
    );

    // Check for critical resource warnings
    expect(
      screen.getByText(`Critical ${ensureStringResourceType(ResourceType.MINERALS)} levels`)
    ).toBeInTheDocument();
    expect(
      screen.getByText(`Critical ${ensureStringResourceType(ResourceType.ENERGY)} levels`)
    ).toBeInTheDocument();
  });
});
