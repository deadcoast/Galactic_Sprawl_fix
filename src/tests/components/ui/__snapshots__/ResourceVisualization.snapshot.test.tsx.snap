// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ResourceVisualization Component Snapshots > should render correctly with default state 1`] = `
<div>
  <div
    class="grid grid-cols-2 gap-4"
  >
    <div
      class="p-3 bg-amber-900/20 rounded-lg border border-amber-700/30"
    >
      <div
        class="mb-2 flex items-center space-x-3"
      >
        <div
          class="rounded-lg p-1.5 bg-amber-900/20"
        >
          <svg
            class="lucide lucide-database h-4 w-4 text-amber-400"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <ellipse
              cx="12"
              cy="5"
              rx="9"
              ry="3"
            />
            <path
              d="M3 5V19A9 3 0 0 0 21 19V5"
            />
            <path
              d="M3 12A9 3 0 0 0 21 12"
            />
          </svg>
        </div>
        <div
          class="flex-1"
        >
          <div
            class="text-sm font-medium capitalize text-gray-300"
          >
            minerals
          </div>
          <div
            class="text-lg font-bold text-amber-400"
          >
            1,000
             / 10,000
          </div>
        </div>
      </div>
      <div
        class="h-1.5 overflow-hidden rounded-full bg-gray-800"
      >
        <div
          class="h-full bg-amber-500 transition-all duration-500"
          style="opacity: 0.7; width: 0px;"
        />
      </div>
      <div
        class="mt-2 text-xs text-yellow-400"
        style="opacity: 0; height: 0px;"
      >
        Low
         
        minerals
         levels
      </div>
    </div>
    <div
      class="p-3 bg-cyan-900/20 rounded-lg border border-cyan-700/30"
    >
      <div
        class="mb-2 flex items-center space-x-3"
      >
        <div
          class="rounded-lg p-1.5 bg-cyan-900/20"
        >
          <svg
            class="lucide lucide-zap h-4 w-4 text-cyan-400"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polygon
              points="13 2 3 14 12 14 11 22 21 10 12 10 13 2"
            />
          </svg>
        </div>
        <div
          class="flex-1"
        >
          <div
            class="text-sm font-medium capitalize text-gray-300"
          >
            energy
          </div>
          <div
            class="text-lg font-bold text-cyan-400"
          >
            1,000
             / 8,000
          </div>
        </div>
      </div>
      <div
        class="h-1.5 overflow-hidden rounded-full bg-gray-800"
      >
        <div
          class="h-full bg-cyan-500 transition-all duration-500"
          style="opacity: 1; width: 0px;"
        />
      </div>
    </div>
    <div
      class="p-3 bg-green-900/20 rounded-lg border border-green-700/30"
    >
      <div
        class="mb-2 flex items-center space-x-3"
      >
        <div
          class="rounded-lg p-1.5 bg-green-900/20"
        >
          <svg
            class="lucide lucide-users h-4 w-4 text-green-400"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 21v-2a4 4 0 0 0-4-4H6a4 4 0 0 0-4 4v2"
            />
            <circle
              cx="9"
              cy="7"
              r="4"
            />
            <path
              d="M22 21v-2a4 4 0 0 0-3-3.87"
            />
            <path
              d="M16 3.13a4 4 0 0 1 0 7.75"
            />
          </svg>
        </div>
        <div
          class="flex-1"
        >
          <div
            class="text-sm font-medium capitalize text-gray-300"
          >
            population
          </div>
          <div
            class="text-lg font-bold text-green-400"
          >
            100
          </div>
        </div>
      </div>
      <div
        class="h-1.5 overflow-hidden rounded-full bg-gray-800"
      >
        <div
          class="h-full bg-green-500 transition-all duration-500"
          style="opacity: 1; width: 0px;"
        />
      </div>
    </div>
    <div
      class="p-3 bg-purple-900/20 rounded-lg border border-purple-700/30"
    >
      <div
        class="mb-2 flex items-center space-x-3"
      >
        <div
          class="rounded-lg p-1.5 bg-purple-900/20"
        >
          <svg
            class="lucide lucide-beaker h-4 w-4 text-purple-400"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4.5 3h15"
            />
            <path
              d="M6 3v16a2 2 0 0 0 2 2h8a2 2 0 0 0 2-2V3"
            />
            <path
              d="M6 14h12"
            />
          </svg>
        </div>
        <div
          class="flex-1"
        >
          <div
            class="text-sm font-medium capitalize text-gray-300"
          >
            research
          </div>
          <div
            class="text-lg font-bold text-purple-400"
          >
            0
          </div>
        </div>
      </div>
      <div
        class="h-1.5 overflow-hidden rounded-full bg-gray-800"
      >
        <div
          class="h-full bg-purple-500 transition-all duration-500"
          style="opacity: 1; width: 0px;"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`ResourceVisualization Component Snapshots > should render with critical resource levels 1`] = `
<div>
  <div
    class="grid grid-cols-2 gap-4"
  >
    <div
      class="p-3 bg-amber-900/20 rounded-lg border border-amber-700/30"
    >
      <div
        class="mb-2 flex items-center space-x-3"
      >
        <div
          class="rounded-lg p-1.5 bg-amber-900/20"
        >
          <svg
            class="lucide lucide-database h-4 w-4 text-amber-400"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <ellipse
              cx="12"
              cy="5"
              rx="9"
              ry="3"
            />
            <path
              d="M3 5V19A9 3 0 0 0 21 19V5"
            />
            <path
              d="M3 12A9 3 0 0 0 21 12"
            />
          </svg>
        </div>
        <div
          class="flex-1"
        >
          <div
            class="text-sm font-medium capitalize text-gray-300"
          >
            minerals
          </div>
          <div
            class="text-lg font-bold text-amber-400"
          >
            1,000
             / 10,000
          </div>
        </div>
      </div>
      <div
        class="h-1.5 overflow-hidden rounded-full bg-gray-800"
      >
        <div
          class="h-full bg-amber-500 transition-all duration-500"
          style="opacity: 0.7; width: 0px;"
        />
      </div>
      <div
        class="mt-2 text-xs text-yellow-400"
        style="opacity: 0; height: 0px;"
      >
        Low
         
        minerals
         levels
      </div>
    </div>
    <div
      class="p-3 bg-cyan-900/20 rounded-lg border border-cyan-700/30"
    >
      <div
        class="mb-2 flex items-center space-x-3"
      >
        <div
          class="rounded-lg p-1.5 bg-cyan-900/20"
        >
          <svg
            class="lucide lucide-zap h-4 w-4 text-cyan-400"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polygon
              points="13 2 3 14 12 14 11 22 21 10 12 10 13 2"
            />
          </svg>
        </div>
        <div
          class="flex-1"
        >
          <div
            class="text-sm font-medium capitalize text-gray-300"
          >
            energy
          </div>
          <div
            class="text-lg font-bold text-cyan-400"
          >
            1,000
             / 8,000
          </div>
        </div>
      </div>
      <div
        class="h-1.5 overflow-hidden rounded-full bg-gray-800"
      >
        <div
          class="h-full bg-cyan-500 transition-all duration-500"
          style="opacity: 1; width: 0px;"
        />
      </div>
    </div>
    <div
      class="p-3 bg-green-900/20 rounded-lg border border-green-700/30"
    >
      <div
        class="mb-2 flex items-center space-x-3"
      >
        <div
          class="rounded-lg p-1.5 bg-green-900/20"
        >
          <svg
            class="lucide lucide-users h-4 w-4 text-green-400"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 21v-2a4 4 0 0 0-4-4H6a4 4 0 0 0-4 4v2"
            />
            <circle
              cx="9"
              cy="7"
              r="4"
            />
            <path
              d="M22 21v-2a4 4 0 0 0-3-3.87"
            />
            <path
              d="M16 3.13a4 4 0 0 1 0 7.75"
            />
          </svg>
        </div>
        <div
          class="flex-1"
        >
          <div
            class="text-sm font-medium capitalize text-gray-300"
          >
            population
          </div>
          <div
            class="text-lg font-bold text-green-400"
          >
            100
          </div>
        </div>
      </div>
      <div
        class="h-1.5 overflow-hidden rounded-full bg-gray-800"
      >
        <div
          class="h-full bg-green-500 transition-all duration-500"
          style="opacity: 1; width: 0px;"
        />
      </div>
    </div>
    <div
      class="p-3 bg-purple-900/20 rounded-lg border border-purple-700/30"
    >
      <div
        class="mb-2 flex items-center space-x-3"
      >
        <div
          class="rounded-lg p-1.5 bg-purple-900/20"
        >
          <svg
            class="lucide lucide-beaker h-4 w-4 text-purple-400"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4.5 3h15"
            />
            <path
              d="M6 3v16a2 2 0 0 0 2 2h8a2 2 0 0 0 2-2V3"
            />
            <path
              d="M6 14h12"
            />
          </svg>
        </div>
        <div
          class="flex-1"
        >
          <div
            class="text-sm font-medium capitalize text-gray-300"
          >
            research
          </div>
          <div
            class="text-lg font-bold text-purple-400"
          >
            0
          </div>
        </div>
      </div>
      <div
        class="h-1.5 overflow-hidden rounded-full bg-gray-800"
      >
        <div
          class="h-full bg-purple-500 transition-all duration-500"
          style="opacity: 1; width: 0px;"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`ResourceVisualization Component Snapshots > should render with low resource levels 1`] = `
<div>
  <div
    class="grid grid-cols-2 gap-4"
  >
    <div
      class="p-3 bg-amber-900/20 rounded-lg border border-amber-700/30"
    >
      <div
        class="mb-2 flex items-center space-x-3"
      >
        <div
          class="rounded-lg p-1.5 bg-amber-900/20"
        >
          <svg
            class="lucide lucide-database h-4 w-4 text-amber-400"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <ellipse
              cx="12"
              cy="5"
              rx="9"
              ry="3"
            />
            <path
              d="M3 5V19A9 3 0 0 0 21 19V5"
            />
            <path
              d="M3 12A9 3 0 0 0 21 12"
            />
          </svg>
        </div>
        <div
          class="flex-1"
        >
          <div
            class="text-sm font-medium capitalize text-gray-300"
          >
            minerals
          </div>
          <div
            class="text-lg font-bold text-amber-400"
          >
            1,000
             / 10,000
          </div>
        </div>
      </div>
      <div
        class="h-1.5 overflow-hidden rounded-full bg-gray-800"
      >
        <div
          class="h-full bg-amber-500 transition-all duration-500"
          style="opacity: 0.7; width: 0px;"
        />
      </div>
      <div
        class="mt-2 text-xs text-yellow-400"
        style="opacity: 0; height: 0px;"
      >
        Low
         
        minerals
         levels
      </div>
    </div>
    <div
      class="p-3 bg-cyan-900/20 rounded-lg border border-cyan-700/30"
    >
      <div
        class="mb-2 flex items-center space-x-3"
      >
        <div
          class="rounded-lg p-1.5 bg-cyan-900/20"
        >
          <svg
            class="lucide lucide-zap h-4 w-4 text-cyan-400"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polygon
              points="13 2 3 14 12 14 11 22 21 10 12 10 13 2"
            />
          </svg>
        </div>
        <div
          class="flex-1"
        >
          <div
            class="text-sm font-medium capitalize text-gray-300"
          >
            energy
          </div>
          <div
            class="text-lg font-bold text-cyan-400"
          >
            1,000
             / 8,000
          </div>
        </div>
      </div>
      <div
        class="h-1.5 overflow-hidden rounded-full bg-gray-800"
      >
        <div
          class="h-full bg-cyan-500 transition-all duration-500"
          style="opacity: 1; width: 0px;"
        />
      </div>
    </div>
    <div
      class="p-3 bg-green-900/20 rounded-lg border border-green-700/30"
    >
      <div
        class="mb-2 flex items-center space-x-3"
      >
        <div
          class="rounded-lg p-1.5 bg-green-900/20"
        >
          <svg
            class="lucide lucide-users h-4 w-4 text-green-400"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 21v-2a4 4 0 0 0-4-4H6a4 4 0 0 0-4 4v2"
            />
            <circle
              cx="9"
              cy="7"
              r="4"
            />
            <path
              d="M22 21v-2a4 4 0 0 0-3-3.87"
            />
            <path
              d="M16 3.13a4 4 0 0 1 0 7.75"
            />
          </svg>
        </div>
        <div
          class="flex-1"
        >
          <div
            class="text-sm font-medium capitalize text-gray-300"
          >
            population
          </div>
          <div
            class="text-lg font-bold text-green-400"
          >
            100
          </div>
        </div>
      </div>
      <div
        class="h-1.5 overflow-hidden rounded-full bg-gray-800"
      >
        <div
          class="h-full bg-green-500 transition-all duration-500"
          style="opacity: 1; width: 0px;"
        />
      </div>
    </div>
    <div
      class="p-3 bg-purple-900/20 rounded-lg border border-purple-700/30"
    >
      <div
        class="mb-2 flex items-center space-x-3"
      >
        <div
          class="rounded-lg p-1.5 bg-purple-900/20"
        >
          <svg
            class="lucide lucide-beaker h-4 w-4 text-purple-400"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4.5 3h15"
            />
            <path
              d="M6 3v16a2 2 0 0 0 2 2h8a2 2 0 0 0 2-2V3"
            />
            <path
              d="M6 14h12"
            />
          </svg>
        </div>
        <div
          class="flex-1"
        >
          <div
            class="text-sm font-medium capitalize text-gray-300"
          >
            research
          </div>
          <div
            class="text-lg font-bold text-purple-400"
          >
            0
          </div>
        </div>
      </div>
      <div
        class="h-1.5 overflow-hidden rounded-full bg-gray-800"
      >
        <div
          class="h-full bg-purple-500 transition-all duration-500"
          style="opacity: 1; width: 0px;"
        />
      </div>
    </div>
  </div>
</div>
`;
